versioned_docs/version-2.0/self_hosting/configuration/ttl.mdx

```diff
@@ -0,0 +1,37 @@
+import {
+  CodeTabs,
+  DockerBlock,
+  HelmBlock,
+} from "../../../../src/components/InstructionsWithCode";
+
+# TTL and Data Retention
+
+LangSmith Self-Hosted allows enablement of automatic TTL and Data Retention of traces. This can be useful if you're complying with data privacy regulations, or if you want to have more efficient space usage and auto cleanup of your traces.
+Traces can will also be auto-upgraded retention levels based on certain actions or run rule applications. For more details on Data Retention, take a look at the [data retention guide](/concepts/usage_and_billing/data_retention_billing).
+
+## Requirements
+
+You can configure retention through helm or environment variable settings. There are a few options that are
+configurable:
+
+- _Enabled:_ Whether data retention is enabled or disabled. If enabled, via the UI you can your default organization and project TTL tiers to apply to traces (see [data retention guide](/concepts/usage_and_billing/data_retention_billing) for details).
+- _Retention Periods:_ You can configure system-wide retention period for shortlived and longlived traces.
```

Comment:
Do we want to call out that if you set longlived data retention via config, then you can also manage this at the project level?

Also, do we need to change the UI to say the actual customized retention length instead of 14d / 400d? cc @eric-langchain

Reply:
Updated.  Also here is the PR for the FE https://github.com/langchain-ai/langchainplus/pull/4665